integration_tests/src/main/python/date_time_test.py

```diff
@@ -460,6 +460,15 @@ def test_to_timestamp(parser_policy):
         { "spark.sql.legacy.timeParserPolicy": parser_policy})
 
 
+def test_to_timestamp_legacy_mode_yyyyMMdd_format():
+    gen = StringGen("[0-9]{3}[1-9](0[1-9]|1[0-2])(0[1-9]|[1-2][0-9])")
```

Comment:
Any reason that the last number of the year can not be 0 ?

Reply:
For year 0000, got the following diff, is investigating:
```
-Row(to_timestamp(a, yyyyMMdd)=None)
-Row(to_timestamp(a, yyyyMMdd)=None)
+Row(to_timestamp(a, yyyyMMdd)=-62162553600000000)
+Row(to_timestamp(a, yyyyMMdd)=-62139312000000000)
```

Reply:
python does not like year 0, but spark-shell is OK.
```
$SPARK_HOME/bin/pyspark --master local[*] --jars ${PLUGIN_JAR} --conf spark.plugins=com.nvidia.spark.SQLPlugin --conf spark.rapids.sql.enabled=true --conf spark.rapids.sql.explain=ALL --driver-java-options '-ea -Duser.timezone=UTC '

spark.read.parquet("/home/chongg/a.parquet").createOrReplaceTempView("tab")
spark.sql("select c1 from tab").show()
+----------+
|        c1|
+----------+
|0000-01-01|
|0000-01-01|
+----------+
spark.sql("select to_timestamp(c1, 'yyyy-MM-dd') from tab").collect()
  File "/home/chongg/progs/sparks/spark-home/python/pyspark/sql/types.py", line 223, in fromInternal
    return datetime.datetime.fromtimestamp(ts // 1000000).replace(microsecond=ts % 1000000)
ValueError: year -1 is out of range
```

When running in spark-shell, it does not report any error.